integration/test_ml.py

```diff
@@ -0,0 +1,361 @@
+def test_publish_invalid_fails(firebase_model):
+    assert firebase_model.validation_error is not None
+
+    with pytest.raises(exceptions.FailedPreconditionError) as excinfo:
+        ml.publish_model(firebase_model.model_id)
+    check_operation_error(
+        excinfo,
+        'Cannot publish a model that is not verified.')
+
+
+@pytest.mark.parametrize('firebase_model', [FULL_MODEL_ARGS], indirect=True)
+def test_publish_unpublish_non_existing_model(firebase_model):
+    ml.delete_model(firebase_model.model_id)
+
+    with pytest.raises(exceptions.NotFoundError) as excinfo:
+        ml.publish_model(firebase_model.model_id)
+    check_operation_error(
+        excinfo,
+        'Model \'{0}\' was not found'.format(firebase_model.as_dict().get('name')))
+
+    with pytest.raises(exceptions.NotFoundError) as excinfo:
+        ml.unpublish_model(firebase_model.model_id)
+    check_operation_error(
+        excinfo,
+        'Model \'{0}\' was not found'.format(firebase_model.as_dict().get('name')))
+
+
+def test_list_models(model_list):
+    filter_str = 'displayName={0} OR tags:{1}'.format(
+        model_list[0].display_name, model_list[1].tags[0])
+
+    all_models = ml.list_models(list_filter=filter_str)
+    all_model_ids = [mdl.model_id for mdl in all_models.iterate_all()]
+    for mdl in model_list:
+        assert mdl.model_id in all_model_ids
+
+
+def test_list_models_invalid_filter():
+    invalid_filter = 'InvalidFilterParam=123'
+
+    with pytest.raises(exceptions.InvalidArgumentError) as excinfo:
+        ml.list_models(list_filter=invalid_filter)
+    check_firebase_error(excinfo, 400, 'Request contains an invalid argument.')
+
+
+@pytest.mark.parametrize('firebase_model', [NAME_ONLY_ARGS], indirect=True)
+def test_delete_model(firebase_model):
+    ml.delete_model(firebase_model.model_id)
+
+    # Second delete of same model will fail
+    with pytest.raises(exceptions.NotFoundError) as excinfo:
+        ml.delete_model(firebase_model.model_id)
+    check_firebase_error(excinfo, 404, 'Requested entity was not found.')
+
+
+# Test tensor flow conversion functions if tensor flow is enabled.
+#'pip install tensorflow' in the environment if you want _TF_ENABLED = True
+#'pip install tensorflow==2.0.0b' for version 2 etc.
+
+
+def _clean_up_directory(save_dir):
+    if save_dir.startswith(tempfile.gettempdir()) and os.path.exists(save_dir):
+        shutil.rmtree(save_dir)
+
+
+@pytest.fixture
+def keras_model():
+    assert _TF_ENABLED
+    x_array = [-1, 0, 1, 2, 3, 4]
+    y_array = [-3, -1, 1, 3, 5, 7]
+    model = tf.keras.models.Sequential(
+        [tf.keras.layers.Dense(units=1, input_shape=[1])])
+    model.compile(optimizer='sgd', loss='mean_squared_error')
+    model.fit(x_array, y_array, epochs=3)
+    return model
+
+
+@pytest.fixture
+def saved_model_dir(keras_model):
+    assert _TF_ENABLED
+    # different versions have different model conversion capability
```

Comment:
This comment probably should be pushed down a couple of lines.

Also add a comment explaining the requirement that `child` directory must not already exist.

Reply:
done